### DISCUSSION TITLE
Encode quantized DCT arrays as jpeg file?

### DISCUSSION THREAD

**Discussion by Nithanaroy** [eugeneware/jpeg-js#100] [Q&A]
![image](https://user-images.githubusercontent.com/670556/88442461-75669880-cdc9-11ea-9197-c0111a011320.png)
How can I save the quantized DCT arrays as shown above, as a jpeg file using this library?

**Comment by patrickhulce** [comment]
You can't provide quantized DCT input to this library. It only accepts RGBA pixel arrays.

If you've already created quantized DCTs you're already like 90% of the way there anyhow ;)

**Comment by Nithanaroy** [comment]
Thanks @patrickhulce :)
Yeah, can you please guide me how I can apply Huffman encoding using any of the functions in encoder.js or similar? Or any other function / library which I can use or extend to complete the last 10% ?

I extracted the quantized DCT arrays using https://github.com/uber-research/jpeg2dct from a jpeg file. But unable to find an approach to save them as a jpeg.

**Comment by patrickhulce** [comment]
Oh, if a library is outputting DCT for you and you're not manually constructing these in a way you're familiar with, it's probably a bit much to walkthrough and more than 10% to go :/ sorry. 

My best advice is try to find another library that will do this out of the box or dissect the encoder logic in a language you know best.

**Comment by Nithanaroy** [comment]
Thanks @patrickhulce for the tips.
Yeah, I have been trying to find a library that does this (qdct -> .jpeg) but couldn't find any for almost 2 months. I was excited to see this in JS as I'm quite familiar with the language. But due to my limited knowledge in codecs, I was unable to follow the important details in encoder.js to make it work for my use case. I would really appreciate any help here 🙏
